beacon-chain/powchain/log_processing.go

```diff
@@ -424,6 +340,97 @@ func (s *Service) processPastLogs(ctx context.Context) error {
 	return nil
 }
 
+func (s *Service) processBlockInBatch(ctx context.Context, currentBlockNum *uint64, latestFollowHeight uint64, batchSize *uint64, additiveFactor uint64, logCount uint64) error {
```

Comment:
this function changes the behaviour of our headers map. It is meant to be persistent across all calls to request headers across the scope of `processPastLogs` .  Now it just lives in each invocation of `processBlockInBatch` . We should initialize it outside and then pass it in as an argument. 

Also I noticed we are passing in pointers of `currentBlockNum` and `batchSize`. Maybe we can instead return them in `processBlockInBatch` and then set the value in `processPastLogs` ?

Reply:
Yes, you are absolutely right. Thanks for finding my mistake!
I also change it to return `currentBlockNum` and `batchSize` as you mentioned. 

Please take another look.